checker/raw_result.go

```diff
@@ -84,3 +54,60 @@ type Tool struct {
 	Desc          string
 	ConfigFiles   []File
 }
+
+// SecurityPolicyData contains the raw results
+// for the Security-Policy check.
+type SecurityPolicyData struct {
+	// Files contains a list of files.
+	Files []File
+}
+
+// BinaryArtifactData contains the raw results
+// for the Binary-Artifact check.
+type BinaryArtifactData struct {
+	// Files contains a list of files.
+	Files []File
+}
+
+// DependencyUpdateToolData contains the raw results
+// for the Dependency-Update-Tool check.
+type DependencyUpdateToolData struct {
+	// Tools contains a list of tools.
+	// Note: we only populate one entry at most.
+	Tools []Tool
+}
+
+// BranchProtectionData contains the raw results
+// for one branch.
+//nolint:govet
+type BranchProtectionData struct {
+	Protected                           *bool
```

Comment:
This makes me wonder should we follow the same pattern for other raw structs too? That is ptr to primitive types. Can do it in a separate PR if you agree.

Reply:
the pointer `nil` means we could not retrieve the information. Other checks don't have this problem, right?
I've not thought this through, but I'm not sure it would help for the other checks I have migrated, like Security-Policy, Dependency-Tool and Binary-Artifacts. Do you have another idea? Which checks do you you think would benefit from it?

Reply:
> the pointer `nil` means we could not retrieve the information. Other checks don't have this problem, right?

Well, it should be `nil` even if we don't run the check. Which can happen with any check.

IMO it'll be useful to have a consistent convention across checks. Anyways, this is not a blocking issue. We can migrate all checks to the raw/evaluation format and then reconsider this.